patroni/api.py

```diff
@@ -512,7 +512,21 @@ def check_auth_header(self, auth_header):
 
     @staticmethod
     def __get_ssl_options(config):
-        return {option: config[option] for option in ['certfile', 'keyfile'] if option in config}
+        options = {option: config[option] for option in ['certfile', 'keyfile', 'ciphers'] if option in config}
+        if 'ssl_versions' in config:
+            import ssl
+            protocols = {
+                'TLSv1.0': ssl.PROTOCOL_TLSv1,
+                'TLSv1.1': ssl.PROTOCOL_TLSv1_1,
+                'TLSv1.2': ssl.PROTOCOL_TLSv1_2
+            }
+            for version in config['ssl_versions']:
+                if version in protocols:
+                    if 'ssl_version' not in options:
+                        options['ssl_version'] = protocols[version]
+                    else:
+                        options['ssl_version'] |= protocols[version]
```

Comment:
There are a couple of issues with `ssl_version`
1. First of all you can't use bitwise or operator to combine multiple versions
2. The [documentation](https://docs.python.org/2/library/ssl.html#ssl.PROTOCOL_TLSv1) tells that usage of `ssl.PROTOCOL*` is deprecated:
```
Deprecated since version 2.7.13: OpenSSL has deprecated all version specific protocols.
Use the default protocol with flags like OP_NO_SSLv3 instead.
```

Reply:
Thanks for your feedback. To follow python recommendations, I've replaced the direct call to `ssl.wrap_socket`, which is deprecated, with an `SSLContext` with secure settings then a call to `SSLContext.wrap_socket`. The REST API is now secure by default.

Reply:
But now, tests are failing because `ssl.wrap_socket` was mocked. Do I have to provide tests for the context or can we update the patch decorator? In both cases, I need your help, please. I'm not that comfortable with python tests.